### PR TITLE
tailscale: 1.46.1 -> 1.48.1

### LINKED ISSUE #250647: Update request: tailscale 1.46.1 → 1.48.1
- Package name:
- Latest released version:

- Current version on the unstable channel:
- Current version on the stable/release channel:

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by mfrw**
Diff: https://github.com/tailscale/tailscale/compare/v1.46.1...v1.48.1
Changelog: https://github.com/tailscale/tailscale/releases/tag/v1.48.1
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Closes: #250647

### CODE REVIEW

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 250648` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>tailscale</li>
    <li>trayscale</li>
  </ul>
</details>

**Review by twitchyliquid64 — APPROVED** [review]
Excited to have more maintainers! :D

**Review by twitchyliquid64 — APPROVED** [review]

**Review by 06kellyjac — CHANGES_REQUESTED** [review]
Very nice. Welcome aboard @mfrw 

One nitpick is we typically swap out the go version within `all-packages.nix`

I believe the benefit of doing this is it's easier to see all the packages not using default `buildGoModule` in one place when updating but I don't have much of a preference myself

so the bit like this `tailscale = callPackage ../xyz { };` would become something like this

```
tailscale = callPackage ../xyz {
  buildGoModule = buildGo121Module;
};
```

(close approximations)

**Comment by mfrw** [comment]
> Very nice. Welcome aboard @mfrw
> 
> One nitpick is we typically swap out the go version within `all-packages.nix`
> 
> I believe the benefit of doing this is it's easier to see all the packages not using default `buildGoModule` in one place when updating but I don't have much of a preference myself
> 
> so the bit like this `tailscale = callPackage ../xyz { };` would become something like this
> 
> ```
> tailscale = callPackage ../xyz {
>   buildGoModule = buildGo121Module;
> };
> ```
> 
> (close approximations)

Thanks, this looks like a much cleaner approach :)

**Comment by natsukium** [comment]
Would you like to add yourself as a maintainer of the package itself?

**Comment by mfrw** [comment]
> Would you like to add yourself as a maintainer of the package itself?

Yes sir :)

File: pkgs/servers/tailscale/default.nix
Comment (natsukium) [inline pkgs/servers/tailscale/default.nix:1]: You also need to add yourself as a maintainer to this file.

Comment (mfrw) [reply pkgs/servers/tailscale/default.nix:1]: I think I did, Thank you :)

Comment (natsukium) [reply pkgs/servers/tailscale/default.nix:1]: see line 46.
It seems that you have only modified modules.

Comment (mfrw) [reply pkgs/servers/tailscale/default.nix:1]: I am not really sure what happened here.
I have had 4 commits to this file and one commit adds me as a maintainer:
https://github.com/NixOS/nixpkgs/pull/250648/commits/906266e67d9e6cdeddf9f90716fc7d861705f308


This is super strange :(

Comment (mfrw) [reply pkgs/servers/tailscale/default.nix:1]: Done.
Thank you @natsukium for your patience :)

**Review by 06kellyjac — APPROVED** [review]
Ideally `tailscale: use buildGo121Module` can be fixup-ed into the update commit but not a big deal

LGTM thanks

**Review by nshalman — APPROVED** [review]
Looking forward to this landing! Thank you!

**Comment by JeffLabonte** [comment]
Same! Thx for your contribution! Sadly it seems the backport failed ☝️ 😞

**Comment by drupol** [comment]
Hi,

Are you willing to submit a PR to fix this manually ?

If so, please ping me in it, I can review it while my plane is delayed.

**Comment by nshalman** [comment]
Looks like `buildGo121Module` isn't defined back in 2023.05.

**Comment by nshalman** [comment]
#252029 backports 1.46.1 to to 23.05 if we want it.

**Pull request merged**